core/src/main/java/hudson/PluginWrapper.java

```diff
@@ -524,13 +527,40 @@ public boolean hasLicensesXml() {
      */
     /*package*/ void resolvePluginDependencies() throws IOException {
         List<String> missingDependencies = new ArrayList<String>();
+        List<String> obsoleteDependencies = new ArrayList<String>();
         // make sure dependencies exist
         for (Dependency d : dependencies) {
-            if (parent.getPlugin(d.shortName) == null)
+            PluginWrapper dependency = parent.getPlugin(d.shortName);
+            if (dependency == null) {
                 missingDependencies.add(d.toString());
+            }  else {
+                if (!Configuration.getBooleanConfigParameter(DISABLE_PLUGIN_DEPENDENCIES_VERSION_CHECK, false)) {
+                    if (dependency.getVersionNumber().isOlderThan(new VersionNumber(d.version))) {
+                        obsoleteDependencies.add(dependency.getShortName() + "(" + dependency.getVersion() + " < " + d.version + ")");
+                    }
+                }
+            }
+        }
+        StringBuilder messageBuilder = new StringBuilder();
+        if (!missingDependencies.isEmpty()) {
+            boolean plural = missingDependencies.size() > 1;
+            messageBuilder.append(plural ? "Dependencies " : "Dependency ")
+                    .append(Util.join(missingDependencies, ", "))
+                    .append(" ").append(plural ? "don't" : "doesn't")
+                    .append(" exist. ");
+        }
+        if (!obsoleteDependencies.isEmpty()) {
+            boolean plural = obsoleteDependencies.size() > 1;
+            messageBuilder.append(plural ? "Dependencies " : "Dependency ")
+                    .append(Util.join(obsoleteDependencies, ", "))
+                    .append(" ").append(plural ? "are" : "is")
+                    .append(" older than required.").append(" To disable the version check, use -D")
```

Comment:
Not sure this should be advertised so aggressively. We may well remove support for it at some point.